it_hardware/data/ir_model_fields.xml

```diff
@@ -0,0 +1,20 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+    <record id="new_date_lot_serial_x_warranty_date" model="ir.model.fields">
+        <field name="name">x_warranty_date</field>
+        <field name="model_id" ref="stock.model_stock_lot"/>
+        <field name="ttype">date</field>
+        <field name="field_description">Warranty Date</field>
+        <field name="compute"><![CDATA[
+for lot in self:
+    lot.x_warranty_date = fields.Date.today() + timedelta(days=lot.product_id.expiration_time)
```

Comment:
Ok, now I get it. This won't work, as the warranty date will be extended anytime the lot is updated (and that is not what we want).
And there is no field to base this compute on (or maybe `last_delivery_partner_id`? :thinking: but we would need to track on unique serial number)
So my conclusion would be that we use a server action (instead of a compute), as done in solar_installation.
Sorry for the bad idea :shrug:

Reply:
I also think we shouldn't use `expiration_time`, as it has its own meaning and dependencies. I'd add a new field, `x_warranty_days`, on `product.template` that would define the duration of the warranty per product